handlers/whatsapp/whatsapp.go

```diff
@@ -863,7 +864,14 @@ func (h *handler) fetchMediaID(msg courier.Msg, mimeType, mediaURL string) (stri
 		return "", logs, errors.Wrapf(err, "error building request to media endpoint")
 	}
 	setWhatsAppAuthHeader(&req.Header, msg.Channel())
-	req.Header.Add("Content-Type", http.DetectContentType(rr.Body))
+	mtype := http.DetectContentType(rr.Body)
```

Comment:
is there any reason to call this first rather than just `mimetype.Detect(rr.Body)` ?

Reply:
This is the current library, the intention would be not to stop using it, but to add more support with the new library.

Reply:
Yes but what is the point of calling `http.DetectContentType(rr.Body)` first? Is there a case where it returns something correct that `mimetype.Detect(rr.Body)` doesn't ?

Reply:
Not that I have identified, I left it before because it is the current library and works well for other cases.

Reply:
I would assume that this new library detects a superset of things the built-in `http` function does, in which case let's just use it.

Reply:
Ok I added https://github.com/nyaruka/gocommon/pull/60 with some tests to see what the differences are - so let's update to gocommon v1.18.0 and anywhere we're calling `http.DetectContentType` we can call `httpx.DetectContentType` instead

Reply:
Ok, so I update this pull request, or would you prefer I open another one with the `gocommon v1.18.0` update and the function replacement for `httpx.DetectContentType`?

Reply:
Whatever is easiest. I already made the change in mailroom in https://github.com/nyaruka/mailroom/commit/dffc4980bcf948f5c36ee12d1869a0cbdc6d9164